x-pack/plugins/apm/server/routes/traces/get_top_traces.ts

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { sortBy } from 'lodash';
+import moment from 'moment';
+import {
+  SERVICE_NAME,
+  TRANSACTION_NAME,
+} from '../../../common/elasticsearch_fieldnames';
+import { joinByKey } from '../../../common/utils/join_by_key';
+import { withApmSpan } from '../../utils/with_apm_span';
+import { Setup } from '../../lib/helpers/setup_request';
+import { getAverages, getCounts, getSums } from './get_transaction_group_stats';
+import { AgentName } from '../../../typings/es_schemas/ui/fields/agent';
+
+type Key = Record<'service.name' | 'transaction.name', string>;
+export interface TransactionGroup {
+  key: Key;
+  serviceName: string;
+  transactionName: string;
+  transactionType: string;
+  averageResponseTime: number | null | undefined;
+  transactionsPerMinute: number;
+  impact: number;
+  agentName: AgentName;
+}
+
+export type TransactionGroupSetup = Setup;
```

Comment:
Do we need this type if it's just a rename of `Setup`?

```suggestion
```

Reply:
good catch. 99.9% we don't need it